### ISSUE TITLE
Unable to run migrations while trying to upgrade

### ISSUE THREAD

**Issue by pshushereba** [Nodewood/issues#7]
I've been trying to upgrade my Nodewood project from Vue 2 to Vue 3. 

I started by trying to update my Nodewood CLI to 1.0.2, but ran into this error:

<img width="1000" alt="Screen Shot 2022-09-17 at 10 29 05 AM" src="https://user-images.githubusercontent.com/13125407/190869388-8b4639ff-b838-4936-9698-82662afc25e7.png">

If I run `nodewood migrate`, I get this error:
<img width="997" alt="Screen Shot 2022-09-17 at 10 29 23 AM" src="https://user-images.githubusercontent.com/13125407/190869435-55f0ef0a-edb0-4697-8e7e-d5d1feb1d03b.png">

Is there a step that I'm missing here? I was looking for an upgrade guide, but couldn't find anything definitive. I was following the steps [here](https://nodewood.com/docs/getting-started/installation/). I was starting from step 5 because I had done the previous steps with the initial setup.

Here is my root level `package.json`:

```json
{
  "name": "indie-audio",
  "version": "0.1.0",
  "description": "",
  "main": "app/api/api.js",
  "private": true,
  "_moduleAliases": {
    "@app": "./app",
    "@wood": "./wood"
  },
  "scripts": {
    "lint": "npx eslint app wood",
    "migrate": "knex migrate:latest",
    "rollback": "knex migrate:rollback",
    "migrate-test": "knex migrate:latest --env test",
    "rollback-test": "knex migrate:rollback --env test",
    "create-migration": "cd app && knex migrate:make",
    "test": "LOG_LEVEL=error npx jest --runInBand --detectOpenHandles",
    "dev-ui": "cd app && vue-cli-service serve",
    "build-ui": "cd app && vue-cli-service build",
    "dev-api": "cd app/api && NODE_ENV=development nodemon --watch . --watch ../config --watch ../features --watch ../lib --watch ../../wood -L ./api.js | npx pino-pretty"
  },
  "devDependencies": {
    "nodemon": "^2.0.7",
    "eslint": "^7.19.0",
    "husky": "^5.0.9",
    "lint-staged": "^10.5.4"
  },
  "workspaces": [
    "app",
    "wood"
  ],
  "husky": {
    "hooks": {
      "pre-commit": "lint-staged && wood/prevent_wood_commit.sh"
    }
  },
  "lint-staged": {
    "**/*.js": "eslint"
  }
}
```

I don't see Knex in the dependencies so it makes sense it's not finding it. But when I `cd` into the `wood` folder, where I know Knex is a dependency, I'm told to run `nodewood migrate` from the root of a Nodewood project.

When I run `nodewood dev` and check Docker, I can see that not everything is running:
<img width="805" alt="Screen Shot 2022-09-17 at 2 19 43 PM" src="https://user-images.githubusercontent.com/13125407/190876822-7dacbb3e-bbfe-40e1-8041-3a631e5b5465.png">

**Comment by DanHulton** [comment]
Hi Patrick,

Sorry you're running into these issues.  I think I know why you're running into these errors, and I'll work on building a longer-term fix into the CLI, but I also want to get you unblocked ASAP, so I think I have a workaround for now.

The problem is that the CLI supports renaming all your tailwind classes with a prefix to avoid having collisions, and it gets that prefix from the tailwind config file.  However, it does that before installing all the new packages for the upgrade, which means the new `@tailwind/colors` package isn't installed yet.  So a quick workaround for this should be to temporarily replace your `app/tailwind.config.js` with just:

```
module.exports = {};
```

And then restore it after the upgrade is done.  This will avoid loading the upgraded `wood` version of the config with the invalid `require()` statements during the upgrade.

As for upgrade instructions, it can actually be pretty involved to upgrade from Vue 2 to 3, so there's an upgrade guide that starts here: https://nodewood.com/docs/upgrade-guide/#_0-18-0, where Nodewood upgraded to Vue 3 in compatibility mode, then progressed to https://nodewood.com/docs/upgrade-guide/#_0-20-0, where compatibility mode was disabled.  The latter link also has instructions on how to re-enable compatibility mode, in case there are plugins or libraries you're relying on that haven't been updated for Vue 3 or that you can't find replacements for.

What version of Nodewood are you upgrading from?  There may be a significant number of steps from that upgrade guide that you have to go through, and it might actually be easier to start with a fresh install of Nodewood 1.0, copy your `features` folder into it, add in the npm libraries you need, and address the remaining inconsistencies from there.

Regardless, I'm here to help through the process!  Please let me know how it goes and if you run into any more issues.

**Comment by pshushereba** [comment]
Thank you for getting back to me! I can make that change and see if that unblocks me. I am upgrading from 0.17.1. I don't mind running through the steps in each part of the upgrade guides from the previous releases. As long as I know that's what I need to do. I will say however, that currently I don't have anything worth preserving in terms of business logic. So if it's easier to go with the second route you mentioned that's no big deal.

**Comment by DanHulton** [comment]
Hi, yes, it is absolutely easier to go that second route, and start from a fresh Nodewood project.  Pre-1.0 releases tended to have a lot change from release to release, and from 0.17 to 1.0 there are a few "speedbump" releases that are likely to eat up a significant chunk of your time.  If you have minimal logic to copy across, starting fresh is the way to go.

**Comment by pshushereba** [comment]
Awesome. FWIW the fix you suggested worked for me. I was able to get up and running. As far as creating a fresh project, what does that look like? Am I able to just run `nodewood new` and then reuse my API key? Or do I need to generate a new key and then run through the process?

**Comment by DanHulton** [comment]
You can safely re-use the Project API key when recreating your project, yes.  If your license only allows for one project, you're legally only allowed to use your one generated API key for that one project, but there aren't any technical restrictions to enforce it further.

**Comment by pshushereba** [comment]
Great, thank you! I was able to create another project, and I think it has all the updated versions of Vue and everything else. I'm having a separate problem now with trying to create a user. When I try and sign up, the email is logged to the console (I haven't set up the mailer yet). When I try and click the link, it tells me that `Could not find a user with that creation token.`

The only reason that I bring it up is because I had to regenerate my API keys because I needed a new secret key to create the new project. And I wasn't sure if this issue is related to that. If it's not then I'll gladly open up a new issue in the morning.

**Comment by DanHulton** [comment]
That should be unrelated to the project keys - they're only used with the API to create and update your project.

Please feel free to open a new issues with the details and I'll be happy to investigate there.  One suggestion is to ensure your migrations have been run, and then examine the `users` table in the database.  Your new user should have a `secure_flags` JSONB field with a `creationToken` value.  This value should be in the link generated in the console.  If not, please note this in the new ticket.  Also, please confirm that the full value of the token is correctly transferred to your browser when clicking the link.  It's possible that your terminal is not correctly detecting the entire URL and is cutting off part of the token.

**Comment by pshushereba** [comment]
Thank you! I'm going to close this issue out since it's resolved. I'll keep tinkering, and if I'm still having trouble tomorrow I'll open a new issue.
